lib/documents.js

```diff
@@ -140,10 +144,18 @@ export async function register({
     // 3. Attempt to refresh the expiration for the document if it is not
     //    known to be a new registration.
     if(!newRegistration &&
-      await _refresh({externalIdHash, documentHash, ttl, creatorHash})) {
+      await _refresh({
+        externalIdHash, documentHash, ttl, creatorHash,
+        minAssuranceForResolution
+      })) {
+      const {registration} =
+        await _getRegistrationRecord({externalIdHash, documentHash});
+      await updateBatch({
```

Comment:
So this is the hottest code path. We need the flag here we discussed that determines whether or signals whether or not batches should be updated so that we can only very rarely execute these two calls. Also, it's potentially more than one batch that gets updated, I believe.